app/oracle.py

```diff
@@ -266,4 +285,17 @@ def prompt(prompt_message, prompt_end):
         break
 
     logging.info(f'We are in DAEMON mode. Sleep {await_time_in_sec} s and continue')
-    time.sleep(await_time_in_sec)
+
+    # sleep and countdown
+    awake_at = time.time() + await_time_in_sec
+    while time.time() < awake_at:
```

Comment:
I think the main metric update be done in the main loop to reuse calculations done there.
 
getPreviousMetrics are easy on the node so can be done at any time but, alas, changes only once in 24h.
getCurrentMetrics is very heavy load, it's better to reuse the crunched numbers.

I can see simple metrics like "current block of eth1/current slot of eth2" exported in this loop though.

Reply:
done, I removed recalculation of get_current_metrics, get_previous_metrics